src/pages/keyword-generator/KeywordGenerator.vue

```diff
@@ -1,9 +1,192 @@
-<script setup>
-</script>
 <template>
-    <div class="ma-keywords-generator">
-        <div class="ma-header">
+  <div class="ma-keywords-generator">
+    <div class="card">
+      <div class="container">
+        <div :class="['input-container', { 'centered': Object.keys(keywords).some(key => keywords[key].length) }]">
+          <div class="ma-header">
             <span>Keyword Generator</span>
+          </div>
+          <textarea ref="textarea" v-model="inputValue" placeholder="Enter text here..." rows="6" class="input-textarea"></textarea>
+          <button @click="generateKeywords">Generate Keywords</button>
+        </div>
+        <div v-if="Object.keys(keywords).some(key => keywords[key].length)" class="keywords-output">
+          <div class="ma-header">
+            <span>Generated Keywords</span>
+          </div>
+          <div class="filter-container">
+            <label for="ngram-filter">Filter by:</label>
+            <select id="ngram-filter" v-model="selectedNGram" @change="filterKeywords">
+              <option v-for="(value, key) in keywords" :key="key" :value="key">{{ key }}</option>
+            </select>
+          </div>
+          <div>
+            <h4>{{ getNGramLabel() }}</h4>
+            <ul>
+              <li v-for="(item, index) in filteredKeywords" :key="index">{{ item }}</li>
+            </ul>
+          </div>
         </div>
+      </div>
     </div>
-</template>
\ No newline at end of file
+  </div>
+</template>
+
+<script setup>
+import { ref, computed } from 'vue';
+
+const inputValue = ref('');
+const keywords = ref({});
```

Comment:
The variable name `keywords` reminds me of something like: `['foo', 'bar', ...]`

```suggestion
const nGrams = ref({});
```